Cairo Desktop/CairoDesktop.AppGrabber/TaskCategoryProvider.cs

```diff
@@ -59,6 +59,14 @@ public void SetCategoryChangeDelegate(TaskCategoryChangeDelegate changeDelegate)
             categoryChangeDelegate = changeDelegate;
             _appGrabber.CategoryList.CategoryChanged += (sender, args) => categoryChangeDelegate();
             _appGrabber.CategoryList.CollectionChanged += CategoryList_CollectionChanged;
+
+            // request new categories in case of preference change
+            // nullify all existing categories so we don't attempt reuse
+            foreach (ApplicationWindow window in _shellManager.Tasks.GroupedWindows)
+            {
+                window.Category = null;
+            }
+            categoryChangeDelegate();
```

Comment:
Should this guard against null delegates? categoryChangeDelegate?.Invoke would also probably be slightly better.